airflow/utils/db.py

```diff
@@ -69,21 +69,43 @@
 
 log = logging.getLogger(__name__)
 
+# The following code is autogenerated by `scripts/ci/pre_commit/pre_commit_revision_map.py`
+# Begin autogenerated REVISION_HEADS_MAP
+
 REVISION_HEADS_MAP = {
-    "2.0.0": "e959f08ac86c",
-    "2.0.1": "82b7c48c147f",
-    "2.0.2": "2e42bb497a22",
-    "2.1.0": "a13f7613ad25",
-    "2.1.1": "a13f7613ad25",
-    "2.1.2": "a13f7613ad25",
-    "2.1.3": "97cdd93827b8",
-    "2.1.4": "ccde3e26fe78",
-    "2.2.0": "7b2661a43ba3",
-    "2.2.1": "7b2661a43ba3",
-    "2.2.2": "7b2661a43ba3",
+    "2.3.0": "c306b5b5ae4a",
+    "2.2.4": "c381b21cb7e4",
     "2.2.3": "be2bfac3da23",
+    "2.2.0": "7b2661a43ba3",
```

Comment:
good catch.  i think  we'll need  to make some kind of "revision getter" function that will give us the migration that applies to a given release.  so e.g. if  you ask for `2.2.1`  it will walk back to the prior semver number and give that revision..... in a sense "filling in the gaps" .... that should work  yeah?

Reply:
I mean, how would it know that `2.1.7` was never a thing? It could know what the right migration is, but really that should raise instead.

Reply:
(Fwiw, I'm less worries about automating this part, it's a one time thing during the release process)

Reply:
hmm... well ... that  wouldn't necessarily be the responsibility of  this "getter"

the contract for this funciton could be, given a semver number, return the latest revision less than  or equal to that number.

maybe there's somewhere else we could put all known releases.   although this structure could _also_ be used for that  i suppose.

do we have  some way of detecting when  we're in a "release commit"?

then perhaps that is something that could be automated as part of the release process in a precommit

Reply:
other thought....

it's also possible that, just  like how now migrations are stamped with whatever is in `airflow.version`, we could always take whatever is in airflow version and merge it into the "known release list".